lib/ansible/modules/network/avi/avi_api_session.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+"""
+# Created on Aug 12, 2016
+#
+# @author: Gaurav Rastogi (grastogi@avinetworks.com) GitHub ID: grastogi23
+#
+# module_check: not supported
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+
+import json
+import time
+from ansible.module_utils.basic import AnsibleModule
+from copy import deepcopy
+
+HAS_AVI = True
+try:
+    from avi.sdk.avi_api import ApiSession, ObjectNotFound
+    from avi.sdk.utils.ansible_utils import (
+        ansible_return, avi_obj_cmp, cleanup_absent_fields,
+        avi_common_argument_spec)
+except ImportError:
+    HAS_AVI = False
+
+
+DOCUMENTATION = '''
+---
+module: avi_api
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: API Module for invoking any Avi API.
+description:
+    - This module is used to configure Pool object
+version_added: 2.3
+requirements: [ avisdk ]
+options:
+    http_method:
+        description:
+            - Allowed HTTP methods for RESTful services and are supported by
+            Avi Controller.
```

Comment:
This needs to be a single line, it's currently invalid YAML and causing the following shippable error.
```
2017-01-18 19:04:21 ============================================================================
2017-01-18 19:04:21 lib/ansible/modules/network/avi/avi_api_session.py
2017-01-18 19:04:21 ============================================================================
2017-01-18 19:04:21 TRACE:
2017-01-18 19:04:21     while scanning a simple key
2017-01-18 19:04:21       in "<string>", line 15, column 13:
2017-01-18 19:04:21                     Avi Controller.
2017-01-18 19:04:21                     ^
2017-01-18 19:04:21     could not find expected ':'
2017-01-18 19:04:21       in "avi_api_session.DOCUMENTATION", line 56, column 9:
2017-01-18 19:04:21                 choices: ["get", "put", "post",  ... 
2017-01-18 19:04:21                 ^
2017-01-18 19:04:21 ERROR: DOCUMENTATION is not valid YAML. Line 56 column 9
```